processor/runtime/src/main/java/com/redhat/service/smartevents/processor/actions/source/SourceActionInvoker.java

```diff
@@ -0,0 +1,90 @@
+package com.redhat.service.smartevents.processor.actions.source;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.redhat.service.smartevents.infra.auth.AbstractOidcClient;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.CloudEventDeserializationException;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.CloudEventSerializationException;
+import com.redhat.service.smartevents.processor.actions.ActionInvoker;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+public class SourceActionInvoker implements ActionInvoker {
+
+    public static final String CLOUD_EVENT_SOURCE = "RHOAS";
+
+    private static final Logger LOG = LoggerFactory.getLogger(SourceActionInvoker.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final Supplier<String> UUID_GENERATOR = () -> UUID.randomUUID().toString();
+
+    private final String endpoint;
+    private final String cloudEventType;
+    private final WebClient client;
+    private final AbstractOidcClient oidcClient;
+    private final Supplier<String> idGenerator;
+
+    public SourceActionInvoker(String endpoint, String cloudEventType, WebClient client, AbstractOidcClient oidcClient) {
+        this(endpoint, cloudEventType, client, oidcClient, UUID_GENERATOR);
+    }
+
+    SourceActionInvoker(String endpoint, String cloudEventType, WebClient client, AbstractOidcClient oidcClient, Supplier<String> idGenerator) {
+        this.endpoint = endpoint;
+        this.cloudEventType = cloudEventType;
+        this.client = client;
+        this.oidcClient = oidcClient;
+        this.idGenerator = idGenerator;
+    }
+
+    @Override
+    public void onEvent(String event) {
+        LOG.info("SourceActionInvoker::onEvent | {} - {} | {}", endpoint, cloudEventType, event);
```

Comment:
@kostola Lots of logging in here that I think is a bit noisy and is also printing senstive data into the logs i.e the token. Can we tidy it up a bit please.

Reply:
Cleaned this part :+1: